bson/src/main/org/bson/internal/ProvidersCodecRegistry.java

```diff
@@ -51,40 +53,55 @@ public <T> Codec<T> get(final Class<T> clazz) {
     @Override
     public <T> Codec<T> get(final Class<T> clazz, final List<Type> typeArguments) {
         notNull("typeArguments", typeArguments);
-        isTrueArgument("typeArguments is not empty", !typeArguments.isEmpty());
```

Comment:
This assertion and the similar assertion in `ChildCodecRegistry.get` contradicted the documentation of `ChildCodecRegistry.get`, which explicitly allows empty `typeArguments` (this PR has not changed that documentation). While all our tests happened to pass despite the assertions existing, users can pass empty `typeArguments` and get an unjustified error.